spec/models/suspended_email_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe SuspendedEmail, type: :model do
+  describe 'associations' do
+    it { should belong_to(:email_address).class_name('EmailAddress') }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:digested_base_email) }
+  end
+
+  describe '.generate_email_digest' do
+    it 'generates the correct digest for a given email' do
+      email = 'test@example.com'
+      expected_digest = OpenSSL::Digest.hexdigest('SHA256', 'test@example.com')
```

Comment:
should we match the hexdigest style we use in the source?
```suggestion
      expected_digest = Digest::SHA256.hexdigest('test@example.com')
```